src/main/resources/testchipip/csrc/SimDromajoCosim.cc

```diff
@@ -0,0 +1,114 @@
+#include <vpi_user.h>
+#include <svdpi.h>
+
+#include <stdio.h>
+#include <unistd.h>
+
+#include "dromajo_wrapper.h"
+
+// gotten from elaboration
+#include "dromajo_params.h"
+
+#define MAX_ARGS 24
+#define MAX_STR_LEN 24
+
+dromajo_t *dromajo = 0;
+
+extern "C" int dromajo_init(
+    const char* bootrom_file,
+    const char* dtb_file,
+    const char* binary_file)
+{
+    // setup arguments
+    char *local_argv[MAX_ARGS];
+    char local_argc = 0;
+    char mmio_range[MAX_STR_LEN] = "";
+    char plic_params[MAX_STR_LEN] = "";
+    char clint_params[MAX_STR_LEN] = "";
+
+    local_argv[local_argc] = (char*)"./dromajo";
```

Comment:
Is this really the best way to do this? It seems very verbose/messy.

Reply:
Frankly I don't know. I just quickly mocked this up and didn't spend to much time trying to clean it up since I assumed most people wouldn't touch this code. I agree that it is messy.

Reply:
Can't you just initialize the `argv` array in a single expression? https://overiq.com/c-programming-101/array-of-pointers-to-strings-in-c/

Reply:
I think I can do most of it like that. However, I think there needs to be some fanciness done to the MMIO, PLIC, CLINT ranges. Ill mess around and see what works.